### PR TITLE
Fix protos serde

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: cosmos-abci/abci/build.rs [hunk 1]
```diff
@@ -1,3 +1,7 @@
 fn main() {
-    tonic_build::compile_protos("proto/types.proto").unwrap();
+    tonic_build::configure()
+        .type_attribute(".google.protobuf.Timestamp", "#[derive(serde::Serialize, serde::Deserialize)]")
```
Comment (enfipy) [inline cosmos-abci/abci/build.rs]: Remove this

File: Cargo.toml [hunk 2]
```diff
@@ -77,3 +77,7 @@ substrate-build-script-utils = '2.0.0'
 [features]
 default = []
 runtime-benchmarks = ['node-template-runtime/runtime-benchmarks']
+
+[patch.crates-io]
+prost = { git = "https://github.com/adoriasoft/prost", rev = "389010a0a79ef2a5805f510a76108e531c1fb621" }
+prost-types = { git = "https://github.com/adoriasoft/prost", rev = "389010a0a79ef2a5805f510a76108e531c1fb621" }
```
Comment (enfipy) [inline Cargo.toml:83]: Add eof

File: cosmos-abci/abci/src/grpc/protos.rs [hunk 3]
```diff
@@ -317,3 +249,33 @@ impl crate::ResponseQuery for ResponseQuery {
         self.codespace = v
     }
 }
+
+#[test]
```
Comment (enfipy) [inline cosmos-abci/abci/src/grpc/protos.rs:253]: Add #[cfg(test)] mod

File: Cargo.toml [hunk 4]
```diff
@@ -77,3 +77,7 @@ substrate-build-script-utils = '2.0.0'
 [features]
 default = []
 runtime-benchmarks = ['node-template-runtime/runtime-benchmarks']
+
+[patch.crates-io]
+prost = { git = "https://github.com/adoriasoft/prost", rev = "389010a0a79ef2a5805f510a76108e531c1fb621" }
```
Comment (enfipy) [inline Cargo.toml:82]: Replace " with ' as in rest of cargo tomls

**Review by enfipy — CHANGES_REQUESTED** [review]

File: cosmos-abci/abci/src/grpc/protos.rs (hunk 3 above)
Comment (Mr-Leshiy) [reply cosmos-abci/abci/src/grpc/protos.rs]: fixed

File: Cargo.toml (hunk 4 above)
Comment (Mr-Leshiy) [reply Cargo.toml]: fixed

File: Cargo.toml (hunk 2 above)
Comment (Mr-Leshiy) [reply Cargo.toml]: fixed

File: cosmos-abci/abci/build.rs (hunk 1 above)
Comment (Mr-Leshiy) [reply cosmos-abci/abci/build.rs]: fixed

**Review by enfipy — APPROVED** [review]
LGTM!

**Pull request merged**